internal/bindings/server.go

```diff
@@ -128,8 +128,7 @@ func NewNode(ctx context.Context, id uint64, address string, dir string) (*Node,
 	defer C.free(unsafe.Pointer(cdir))
 
 	if rc := C.dqlite_node_create(cid, caddress, cdir, &server); rc != 0 {
-		errmsg := C.GoString(C.dqlite_node_errmsg(server))
-		return nil, fmt.Errorf("%s", errmsg)
+		return nil, fmt.Errorf("failed to create node")
```

Comment:
We normally start error messages with first letter capitalized in lxd, is this something we can do in go-dqlite too or is there already a precedent for lower case first errors? Thanks

Reply:
Is there another way to get the error message out of dqlite? As otherwise the error itself isn't too useful.

Reply:
Not sure, there are other uncapitalized Errorf calls in that file but that might be an oversight. I'll see what the majority usage in the project is.

Reply:
thanks!

Reply:
Does rc represent an error code we can include in the error message perhaps?

Reply:
> Does rc represent an error code we can include in the error message perhaps?

Indeed, it does.

Reply:
Yes, it does -- there are two possible values in current dqlite I believe. I'll amend to include that information.

Reply:
Thanks

Reply:
> Not sure, there are other uncapitalized Errorf calls in that file but that might be an oversight. I'll see what the majority usage in the project is.

I believe the convention in golang is usually to have uncapitalized error messages, and go-dqlite is following that. The LXD convention presumably comes from libc/strerror (wild guess). YMMV

Reply:
The fact that `C.dqlite_node_errmsg(server)` is not returning a proper error message is a bug. It might be a bit laborious but it'd be nice to fix that, since it's going to provide much better diagnostic than the `rc` number alone.

Reply:
@freeekanayaka I don't think it's a problem with dqlite_node_errmsg, it's just that we're calling it on NULL, since the `struct dqlite_node *` argument is cleared here: https://github.com/canonical/dqlite/blob/54d8a0b67a95249e20a0567a4c5f994914bf1b77/src/server.c#L162

Reply:
Right, since the dqlite_node_create function clears the entire struct on failure, it makes little sense in calling dqlite_node_errmsg on the now NULL struct.